src/core/storage/disk.test.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import {DiskStorage} from "./disk";
+import tmp from "tmp";
+import fs from "fs-extra";
+import {join as pathJoin} from "path";
+// $FlowIgnore[missing-export]
+import {TextEncoder, TextDecoder} from "util";
+
+describe("core/storage/disk", () => {
+  function tmpWithContents(contents: mixed, dir: string = basedir.name) {
+    const name: string = tmp.tmpNameSync({dir});
+    fs.writeFileSync(name, JSON.stringify(contents));
+    const fname = name.split("/").pop();
+    return fname;
+  }
+  const badPath = () => pathJoin(tmp.dirSync().name, "not-a-real-path");
+  const fooDefault = () => ({foo: 1337});
```

Comment:
Curious why these are methods. Maybe to ensure test data isn't crossing over from test to test? The way I'm used to seeing this done is:
```
  let badPath, fooDefault;
  beforeEach(() => {
    badPath = pathJoin(tmp.dirSync().name, "not-a-real-path");
    foodDefault = {foo: 1337};
  });
```
I see advantages to both.

Reply:
I think we've been avoiding using test hooks like this because it makes refactoring more difficult. Debugging test hook behavior can be quite painful sometimes.